subsys/fs/nvs/nvs.c

```diff
@@ -119,9 +457,7 @@ int _nvs_sector_is_used(struct nvs_fs *fs, off_t offset)
 	rc = 0;
 	offset &= ~(fs->sector_size - 1);
 	for (addr = 0; addr < fs->sector_size; addr += sizeof(buf)) {
-		rc = flash_read(fs->flash_device,
-				fs->offset + offset + addr,
-				&buf, sizeof(buf));
+		rc = _nvs_flash_read(fs, offset + addr,	&buf, sizeof(buf));
```

Comment:
Is there any reason why we a reading one byte at a time?
I would suggest to read a multiple, like a word, to reduce the number of calls.